R/importance.R

```diff
@@ -36,131 +39,243 @@
+    colnames(RNGseeds) <- names(X)
+  }
+  res_list <- lapply(res_list, "[[", "result")
+
+  ############################## prep output ###################################
   # importance results ordered by decreasing importance
-  if (importance_metric == "ratio") {
-    res <- data.table(X = names(res_list), risk_ratio = unlist(res_list))
-    return(res[order(-res$risk_ratio)])
-  } else if (importance_metric == "difference") {
-    res <- data.table(X = names(res_list), risk_difference = unlist(res_list))
-    return(res[order(-res$risk_difference)])
+  result <- data.table(covariate = names(X), metric = unlist(res_list))
+  if (!is.null(covariate_groups)) {
+    colnames(result)[1] <- "covariate_group"
+  }
+  result <- result[order(-result$metric), ]
+
+  # name the importance metric appropriately
+  metric_name <- paste0("risk_", importance_metric)
+  if (!is.null(attr(original_eval, "name"))) {
+    metric_name <- gsub("risk", attr(original_eval, "name"), metric_name)
   }
+  colnames(result)[2] <- metric_name
+
+  if (type == "permute") {
+    result <- list("result" = result, "RNGseeds" = RNGseeds)
+  }
+
+  return(result)
 }
 
 #' Variable Importance Plot
 #'
-#' @param x The 2-column \code{data.table} returned by the call to
-#'  \code{importance}, where the first column is the predictors and the second
-#'  col
+#' @param x The 2-column \code{data.table} returned by \code{importance}, where
+#'  the first column is the covariate/groups and the second column is the
+#'  importance metric.
 #' @param nvar The maximum number of predictors to be plotted. Defaults to 30.
-#' @param ... Other parameters passed to \code{\link[graphics]{dotchart}}.
 #'
 #' @return A \code{\link[graphics]{dotchart}} of variable importance.
 #'
-#' @importFrom graphics dotchart
+#' @importFrom ggplot2 ggplot geom_point coord_flip scale_x_discrete labs
+#' @importFrom data.table data.table
 #'
 #' @name importance_plot
 #' @keywords variable importance
 #'
 #' @export
-importance_plot <- function(x, nvar = min(30, nrow(x)), ...) {
+importance_plot <- function(x, nvar = min(30, nrow(x))) {
+  if (!is.data.table(x)) {
+    x <- x$result
+  }
 
   # get the importance metric
-  if (grepl("ratio", colnames(x)[2])) {
-    xlab <- "Risk Ratio"
-  } else if (grepl("difference", colnames(x)[2])) {
-    xlab <- "Risk Difference"
-  }
+  xlab <- colnames(x)[2]
 
-  # sort by decreasing importance
-  x_sorted <- x[order(-x[, 2])]
-  # subset to include most at most nvar
-  x_sorted <- x_sorted[1:(min(nvar, nrow(x_sorted))), ]
   # sort by increasing importance
-  x_sorted <- x_sorted[order(x_sorted[, 2])]
-
-  # make dotchart with most important variables on top
-  # x_sorted[[2]] is importance scores & x_sorted[[1]] is covariate names
-  dotchart(
-    x = x_sorted[[2]], labels = x_sorted[[1]],
-    xlab = xlab, ylab = "",
-    xlim = c(min(x_sorted[[2]]), max(x_sorted[[2]])), ...
+  x <- x[order(-x[, 2]), ]
```

Comment:
if `x` is expected to be a `data.table` at this point (i can't really tell despite the `is.data.table()` call prior to this), then the same comment above about the use of `data.table::setorder()` applies

Reply:
I made the change in L225 since the reorder was based on a column name, but not in L266 as the reorder is based on a column index and I am not sure if that will work. 

I tried to find an example with reorder based on column index, but didn't see one. However, I did find in the [data.table::setorder() documentation](https://www.rdocumentation.org/packages/data.table/versions/1.14.0/topics/setorder) that queries like `x[order(.)]` are optimised internally to use `data.table`'s fast order, so I think it's ok.

Reply:
ok, sounds good -- thanks for being thorough